### DISCUSSION TITLE
Parcel with Knex - require with expression

### DISCUSSION THREAD

**Discussion by quantumew** [parcel-bundler/parcel#5309] [General] [answered]
Parcel appears to be ignoring the line linked below. How do I ensure these dialect files are bundled? I am likely missing something really obvious here, so any input is greatly appreciated.

https://github.com/knex/knex/blob/e37aeaa31c8ef9c1b07d2e4d3ec6607e557d800d/lib/knex.js#L44

**Comment by 101arrowz** [comment]
Dynamic `require`s simply aren't possible for a bundler to resolve. To humans, it's obvious that the bundler should grab everything in the `dialects` directory, but like most bundlers, Parcel can only resolve `require`s with fixed strings. It's one of the many reasons ESM imports are preferred; what they refer to is known at build time.

This would be a cool addition to Parcel, though. *Maybe* it could be possible with some clever parsing.

**Comment by mischnic** [comment, marked as answer]
Duplicate of https://github.com/parcel-bundler/parcel/issues/125
